src/Executables/Examples/SingletonTovProfile/SingletonTovProfile.hpp

```diff
@@ -0,0 +1,207 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cmath>
+#include <cstddef>
+#include <fstream>
+#include <memory>
+#include <pup.h>
+
+#include "AlgorithmSingleton.hpp"
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "ErrorHandling/FloatingPointExceptions.hpp"
+#include "IO/Connectivity.hpp"
+#include "IO/H5/AccessType.hpp"
+#include "IO/H5/CheckH5.hpp"
+#include "IO/H5/Dat.hpp"
+#include "IO/H5/File.hpp"
+#include "IO/H5/Header.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "Parallel/Info.hpp"
+#include "Parallel/InitializationFunctions.hpp"
+#include "Parallel/Invoke.hpp"
+#include "Parallel/Printf.hpp"
+#include "Parallel/RegisterDerivedClassesWithCharm.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/Tov.hpp"
+#include "PointwiseFunctions/EquationsOfState/Tags.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"  // IWYU pragma: keep
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Literals.hpp"
+
+/*!
+ * \page SingletonTovProfileExecutablePage SingletonTovProfile Executable
+ * \tableofcontents
+ * The %SingletonTovProfile executable solves the TOV equations for the
+ * provided equation of state with the parameters given in the
+ * tests/InputFiles/ExampleExecutables/SingletonTovProfile.yaml input file.
+ *
+ * The results of the solution are then written to the output file
+ * specified in the yaml file. A summary of the results are also printed
```

Comment:
Would it be better to write such a summary to the output file? I don't think it's something we can do right now easily, but it would be possible to add to the HDF5 infra in the future. If you think this is a good idea I can file an issue :)

Reply:
I made this choice because it reduces the iteration time if one has summaries printed to the terminal; having to open a file to see the summary would defeat the purpose of the summary. I can add an option instead that can toggle whether the summary is printed to the terminal or not printed at all, is this ok?